### PR TITLE
Fix conflict of connection name in different schemas

### PR DESCRIPTION
**Pull request by sibeliusseraphini**
This happens when two schemas have a field with the same name
that is a reference field

- fix RisingStack/graffiti-mongoose#84

### CODE REVIEW

File: src/type/type.js [hunk 1]
```diff
@@ -193,7 +204,8 @@ export default function getType(graffitiModels, {name, description, fields}, pat
     if (type === 'Array') {
       if (subtype === 'Object') {
         const fields = subfields;
-        graphQLField.type = new GraphQLList(getType(graffitiModels, {name, description, fields}, newPath, rootType));
+        const connectionName = getConnectionName(graphQLType.name, name);
```
Comment (tothandras) [inline src/type/type.js]: These are not connections, but nested objects. Connections are based on references.

File: src/type/type.js [hunk 2]
```diff
@@ -210,7 +222,8 @@ export default function getType(graffitiModels, {name, description, fields}, pat
       }
     } else if (type === 'Object') {
       const fields = subfields;
-      graphQLField.type = getType(graffitiModels, {name, description, fields}, newPath, rootType);
+      const connectionName = getConnectionName(graphQLType.name, name);
```
Comment (tothandras) [inline src/type/type.js]: same here

**Comment by tothandras** [comment]
Thanks for the PR: Tests would be nice too. :)

**Comment by sibeliusseraphini** [comment]
I've updated the PR, but I've couldn't manage to implement the tests yet

**Comment by tothandras** [comment]
@sibeliusseraphini Thank you! I will release it tomorrow.

**Pull request merged**